aspects/schema.go

```diff
@@ -121,17 +144,29 @@ func (s *StorageSchema) parse(raw json.RawMessage) (parser, error) {
 }
 
 func (s *StorageSchema) newTypeSchema(typ string) (parser, error) {
-	// TODO: add any, int, number, bool, array and user-defined types
+	// TODO: add any, int, number, bool and array
 	switch typ {
 	case "map":
 		return &mapSchema{topSchema: s}, nil
 	case "string":
 		return &stringSchema{}, nil
 	default:
+		if typ != "" && typ[0] == '$' {
```

Comment:
could we perhaps also move the if into the helper?

Reply:
I opted not to since doing it would require an error to be intuitive IMO and that adds a bit more complexity for a line that is only repeated twice. We could return `nil, nil` but that feels less readable since we're calling getUserType with something that isn't a valid user type reference, so I'd expect an error